version: "3.9"

volumes:
  postgres_data:
services:
  db:
    image: kartoza/postgis:14-3.3
    volumes:
      - postgres_data:/var/lib/postgresql
    environment:
      - POSTGRES_DB=minisass,gis
      - POSTGRES_USER=docker
      - POSTGRES_PASS=docker
      - ALLOW_IP_RANGE=0.0.0.0/0
      - POSTGRES_MULTIPLE_EXTENSIONS=postgis,hstore,postgis_topology,postgis_raster,pgrouting
    healthcheck:
      test: /usr/bin/pg_isready -u postgres
      interval: 5s
      timeout: 10s
      retries: 30
  django:
    build:
      context: ./deployment/docker
    volumes:
      - ./django_project:/home/web/django_project
    environment:
      - POSTGRES_DB=minisass
      - POSTGRES_USER=docker
      - POSTGRES_PASS=docker
      - DATABASE_HOST=db
      - DATABASE_PORT=5432
      - ALLOWED_HOSTS='*'
      - SERVER_MODE=production
    depends_on:
      - db
    ports:
      - "8080"
  web:
    image: nginx
    volumes:
      - ./sites-available/default.conf:/etc/nginx/conf.d/default.conf:rw
    ports:
      - "80:80"
    depends_on:
      - django
      - postgres




